src/pulselistener/pulselistener/lib/phabricator.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+import collections
+import enum
+import time
+
+import structlog
+from libmozdata.phabricator import PhabricatorAPI
+
+logger = structlog.get_logger(__name__)
+
+PhabricatorPatch = collections.namedtuple('Diff', 'id, phid, patch, base_revision, commits')
+
+
+class PhabricatorBuildState(enum.Enum):
+    Queued = 1
+    Secured = 2
+    Public = 3
+
+
+class PhabricatorBuild(object):
+    '''
+    A Phabricator buildable, triggered by HarborMaster
+    '''
+    def __init__(self, request):
+        self.diff_id = int(request.rel_url.query.get('diff', 0))
+        self.repo_phid = request.rel_url.query.get('repo')
+        self.revision_id = int(request.rel_url.query.get('revision', 0))
+        self.target_phid = request.rel_url.query.get('target')
+        self.state = PhabricatorBuildState.Queued
+
+        if not self.diff_id or not self.repo_phid or not self.revision_id or not self.target_phid:
+            raise Exception('Invalid webhook parameters')
+        assert isinstance(self.revision_id, int), 'Revision should be an integer'
+        assert isinstance(self.target_phid, str), 'Target should be a string'
+        assert self.target_phid.startswith('PHID-HMBT-'), 'Invalid target format'
+
+        # Remote objects loaded by actions below
+        self.revision = None
+        self.reviewers = []
+        self.diff = None
+        self.stack = []
+
+    def __str__(self):
+        return 'Revison {} - {}'.format(self.revision_id, self.target_phid)
+
+
+class PhabricatorActions(object):
```

Comment:
We could merge this too into PhabricatorCodeReview (it's the only user). Or were you thinking something else could be using this in the future?

Reply:
Yes, i wanted to have these shared tool in the library, thinking that we would end up with a code review client & a risk analysis client.